processor/src/main/java/org/mapstruct/ap/internal/model/common/Type.java

```diff
@@ -443,28 +457,32 @@ public boolean isAssignableTo(Type other) {
                 else if ( Executables.isGetterMethod( candidate ) ) {
                         // the current accessor is a getter (no setter available). But still, an add method is according
                     // to the above strategy (SETTER_PREFERRED || ADDER_PREFERRED) preferred over the getter.
-                    Type targetType = typeFactory.getReturnType(
-                            typeFactory.getMethodType( (DeclaredType) typeMirror, candidate )
-                    );
+                    Type targetType = typeFactory.getReturnType( (DeclaredType) typeMirror, candidate );
```

Comment:
Hmm.. is the `typeFactory.getMethodType` not required..? Or is this handled by type factory?